### PR TITLE
Add a way to override l2cap MTU from the app layer

### PR DESCRIPTION
**Pull request by jocelynmass**
Allow the BLE stack to externally query the l2cap mtu size.

### CODE REVIEW

File: nimble/host/src/ble_l2cap_coc.c [hunk 1]
```diff
@@ -156,6 +156,21 @@ ble_l2cap_coc_srv_find(uint16_t psm)
     return srv;
 }
 
+int
+ble_l2cap_coc_server_mtu_update(uint16_t psm, uint16_t new_mtu)
```
Comment (gumbypp) [inline nimble/host/src/ble_l2cap_coc.c:160]: who calls this function?

Comment (jocelynmass) [reply nimble/host/src/ble_l2cap_coc.c:160]: In the app layer, when L2CAP_MTU forth word is called.

```
FORTH_WORD(L2CAP_MTU, "L2CAP_MTU", (mtu -- ), "Set l2cap mtu")
{
    uint32_t mtu = POP(uint32_t);
    cble_ctx_t *cble = cble_get_ctx();
    
    if(mtu > MYNEWT_VAL(BLE_L2CAP_COC_MPS)){
        CILOG(ERROR, BLE, "L2CAP CoC MTU can't exceed max MPS: %dB", MYNEWT_VAL(BLE_L2CAP_COC_MPS));
        return FORTH_ERR;
    }
    NIMBLE_ERROR(ble_l2cap_coc_server_mtu_update(CBLE_L2CAP_PSM, mtu));
    CILOG(INFO, BLE, "L2CAP CoC MTU overriden to %ldB was %ldB", mtu, cble->l2cap_coc_mps);
    cble->l2cap_coc_mps = mtu;
    return FORTH_OK;
}

Comment (gumbypp) [reply nimble/host/src/ble_l2cap_coc.c:160]: got it

File: nimble/host/src/ble_l2cap_sig.c [hunk 2]
```diff
@@ -1470,7 +1470,14 @@ ble_l2cap_sig_coc_reconfig(uint16_t conn_handle, struct ble_l2cap_chan *chans[],
     proc->op = BLE_L2CAP_SIG_PROC_OP_RECONFIG;
     proc->reconfig.cid_cnt = num;
     proc->reconfig.new_mtu = new_mtu;
+#if MYNEWT_VAL(BLE_L2CAP_COC_MPS_CB_ENABLE)
+    // only support one common MTU size for all l2cap channels
+    struct ble_l2cap_event event;
+    event.type = BLE_L2CAP_EVENT_COC_GET_MTU;
+    proc->reconfig.new_mps = chans[0]->cb(&event, NULL);
```
Comment (gumbypp) [inline nimble/host/src/ble_l2cap_sig.c:1477]: Do we need to check for `num >= 1`? Do we need a loop if `num >= 2`?

File: nimble/host/syscfg.yml [hunk 3]
```diff
@@ -118,6 +118,11 @@ syscfg.defs:
             MSYS blocks.
         value: 'MYNEWT_VAL_MSYS_1_BLOCK_SIZE-8'
 
+    BLE_L2CAP_COC_MPS_CB_ENABLE:
+        description: >
+            If this flag is set to 1, MPS value might be defined by an external callback
```
Comment (gumbypp) [inline nimble/host/syscfg.yml:123]: should this be "MPS value is defined by an external callback"?

File: nimble/host/src/ble_l2cap_sig.c (hunk 2 above)
Comment (jocelynmass) [reply nimble/host/src/ble_l2cap_sig.c:1477]: I'll put this as a todo to make this code compatible with multiple l2cap coc channels

**Pull request merged**